framework/include/constraints/MortarConstraintBase.h

```diff
@@ -75,6 +75,11 @@ class MortarConstraintBase : public Constraint,
    */
   const MooseVariable & variable() const override { return *_var; }
 
+  /**
+   * The variable number that this object operates on (pointer).
+   */
+  const MooseVariable * variable_ptr() const { return _var; }
```

Comment:
With the change I recommended in the `subdomainsCovered` method, we shouldn't need this API anymore. But we should add an assertion in the `variable()` method that checks and sees whether `_var` is not null.